Source/qol/stash.cpp

```diff
@@ -537,6 +533,20 @@ void StashStruct::SetPage(unsigned newPage)
 	page = std::min(newPage, CountStashPages - 1);
 }
 
+void StashStruct::NextPage(unsigned offset)
+{
+	SetPage(page + offset);

```

Comment:
Is this the last caller to `SetPage`?
If yes, should we inline `SetPage` and remove the function?

Reply:
How's this version look? I updated prev/nextpage to be more defensive and realised that we can save/load the current stash page to make use of this function. It's entirely backwards compatible with the current stash format, can be dropped if desired.

Reply:
lgtm 🙂 

For saving the active page:
I was also thinking of adding this to the normal settings/options (`diablo.ini`).
But it can also be saved along the normal stash items.
If we want to save it with the stash, doesn't we need to set the `dirty` flag on the stash?

Reply:
I think it's best to save it with the stash archive itself, purely because there's a single player stash and a multiplayer stash which contain different items. Presumably the player will expect to end up on the last used page in each context.

Good point about the dirty flag, I hadn't picked up on that and was relying on modifications like adding/removing an item to trigger saving a page. I think it makes sense to mark dirty on page change directly so it gets saved even with no other changes.

Reply:
When adding to `diablo.ini` we would have needed multiple entries (diablo sp, diablo mp, hellfire sp, hellfire mp).
The only pro I can think about, is that this setting wouldn't be reverted when loading a game in sp. But even this can be good or bad.